krm-functions/lib/vlanalloc/v1alpha1/vlanalloc.go

```diff
@@ -0,0 +1,57 @@
+/*
+ Copyright 2023 The Nephio Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"github.com/nephio-project/nephio/krm-functions/lib/kubeobject"
+	vlanv1alpha1 "github.com/nokia/k8s-ipam/apis/alloc/vlan/v1alpha1"
+)
+
+type VLANAllocation struct {
+	kubeobject.KubeObjectExt[*vlanv1alpha1.VLANAllocation]
+}
+
+// NewFromKubeObject creates a new KubeObjectExt
+// It expects a *fn.KubeObject as input representing the serialized yaml file
+func NewFromKubeObject(o *fn.KubeObject) (*VLANAllocation, error) {
```

Comment:
In the same spirit as above comment, show we just return the base KubeObjectExt from here? 
In fact I am also thinking can the function devs directly invoke kubeobject.NewFromKubeObject[*vlanv1alpha1.VLANAllocation](fn.KubeObject)?

That could mean do we really need this library for indirection?

Reply:
The value is you get explicit type checking for the object and we need this in several libraries. This is the reason to do this. otherwise people need to implement this in each function independently again and again.

Reply:
Thanks @henderiw . Approved.